### PR TITLE
Add MDS059 blockquote-whitespace rule

### PR DESCRIPTION
**Pull request by jeduden**
Implements rule MDS059 to flag and fix blockquote whitespace defects, covering markdownlint rules MD027 and MD028.

## Summary

Adds a new linting rule that enforces proper whitespace formatting in Markdown blockquotes:
- **MD027**: Flags blockquote markers followed by multiple spaces; autofix collapses to single space
- **MD028**: Flags blank lines between adjacent sibling blockquote blocks (flag-only, no autofix)

## Changes

- **New rule implementation** (`internal/rules/blockquotewhitespace/`):
  - `rule.go`: Core logic for MD027 per-line checks with autofix, and MD028 sibling-gap detection via AST traversal
  - `rule_test.go`: Comprehensive test suite covering both defects, edge cases (nested blockquotes, code blocks), and fix behavior

- **Documentation**:
  - Rule README with examples, configuration, and meta-information
  - Fixture tests (good/bad/fixed markdown files) for integration testing
  - Updated markdownlint coverage matrix to reflect MDS059 implementation
  - Updated plan 177 status from in-progress to complete

- **Integration**:
  - Registered rule in `internal/rules/all/all.go` and `internal/integration/rules_test.go`
  - Updated rules index and coverage documentation

## Implementation Details

- MD027 detection uses regex to match blockquote lines and flag `> {2,}` patterns, skipping code blocks
- MD028 detection walks the AST to find adjacent `*ast.Blockquote` siblings separated only by blank lines
- Fix only applies to MD027 (collapsing spaces); MD028 is intentionally not auto-fixed due to ambiguity in renderer behavior
- Respects code block boundaries to avoid false positives inside fenced code blocks

https://claude.ai/code/session_019yMJmrAgHHt3E8bvhXhLm4

### CODE REVIEW

File: internal/rules/blockquotewhitespace/rule.go [hunk 1]
```diff
@@ -0,0 +1,176 @@
...
+	for i, line := range f.Lines {
+		lineNum := i + 1
+		if codeLines[lineNum] || !reBlockquoteLine.Match(line) {
+			lines[i] = string(line)
+			continue
+		}
+		lines[i] = string(reMultiSpace.ReplaceAll(line, []byte("> ")))
+	}
+	return []byte(strings.Join(lines, "\n"))
+}
```
[the remark this answers is not in the document]
Comment (jeduden) [reply internal/rules/blockquotewhitespace/rule.go:156]: `strings.Join(f.Lines, "\n")` is the established pattern for all fixable rules in this codebase — `notrailingspaces`, `blanklinearoundheadings`, and `blanklinearoundfencedcode` all use the same approach. Changing the line-ending strategy would be a cross-cutting concern that applies to every fixer, not just MDS059. No change here.

---
_Generated by [Claude Code](https://claude.ai/code)_

File: internal/rules/MDS059-blockquote-whitespace/good/default.md [hunk 2]
```diff
@@ -0,0 +1,7 @@
+# Title
+
+> This blockquote has one space after the marker.
+
+Some text between the blockquotes.
+
+> Another blockquote, separated by a paragraph.
```
[the remark this answers is not in the document]
Comment (jeduden) [reply internal/rules/MDS059-blockquote-whitespace/good/default.md:7]: Verified — `go run ./cmd/mdsmith check .` passes across all 321 files (0 failures), which includes `good/default.md` and `good/internal-blank.md`. Per the fixture convention, good fixtures don't require explicit frontmatter; they simply must pass all default-enabled rules, which CI confirms via the `mdsmith-check` job.

---
_Generated by [Claude Code](https://claude.ai/code)_
